config/initializers/class_decorator.rb

```diff
@@ -0,0 +1,22 @@
+class ApplicationRecord < ActiveRecord::Base
+  class << self
+    def decorate
+      @_decorator || (@_decorator = decorator_for(self))
+    end
+
+    def decorator_for(klass)
+      decorator = nil
+
+      while decorator.nil? && klass != ApplicationRecord
+        decorator = "#{klass.name}Decorator".safe_constantize
+        klass = klass.superclass
+      end
+
+      decorator
+    end
+  end
+
+  def decorate
+    self.class.decorator_for(self.class).new(self)
+  end
+end
```

Comment:
> Patching something that doesn't belong to you is a bad idea, especially re-opening a class in our lovely project 😄

I would take this a bit further and state that as you move to the REST API, you *can't* patch platform classes, so doing so here might make it harder to move forward.  However, using the decoration pattern above, it is clear which methods are owned by the UI and not the platform, and those can be converted to code that eventually decorates the result of the REST API.